### ISSUE TITLE
GoogleMap iconUrl does not use my custom svg icon on Android

### ISSUE THREAD

**Issue by rs2-davebanguilan** [ionic-team/capacitor-plugins#1085]
### Plugin(s)
`@capacitor/google-maps": "^1.1.0`

### Capacitor Version
```
Latest Dependencies:

  @capacitor/cli: 3.6.0
  @capacitor/core: 3.6.0
  @capacitor/android: 3.6.0
  @capacitor/ios: 3.6.0

Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/cli: 3.6.0
  @capacitor/core: 3.6.0
  @capacitor/android: 3.6.0

[success] Android looking great! 👌
```

### Platform(s)
- Android

### Current Behavior
The application shows the `iconUrl` on web as you can see on the photo below:
![image](https://user-images.githubusercontent.com/107235131/180004631-ee68f426-a8f5-4847-ba6a-97f2dfb46e58.png)

But the iconUrl does not show on Android:
![image](https://user-images.githubusercontent.com/107235131/180004818-391af181-92e9-4551-ab7c-5444d1ca719a.png)

### Expected Behavior
The `iconUrl` should work the same on Android. It should show the same iconUrl that is shown in web.

### Code Reproduction
https://github.com/rs2-davebanguilan/poc-google-map

### Other Technical Details
```
Ionic:

   Ionic CLI                     : 6.19.1 (C:\Users\dave.banguilan\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.1.14
   @angular-devkit/build-angular : 14.0.6
   @angular-devkit/schematics    : 14.0.6
   @angular/cli                  : 14.0.6
   @ionic/angular-toolkit        : 6.1.0

Capacitor:

   Capacitor CLI      : 3.6.0
   @capacitor/android : 3.6.0
   @capacitor/core    : 3.6.0
   @capacitor/ios     : not installed

Utility:

   cordova-res : 0.15.4
   native-run  : 1.6.0

System:

   NodeJS : v16.14.0 (C:\Program Files\nodejs\node.exe)
   npm    : 8.3.1
   OS     : Windows 10
```

I am currently using svg for my icon due to quality issues. I also tried using a png but it still doesn't work.

**Comment by DwieDima** [comment]
on ios and android projects,  ionic resources are bundled inside `public` folder.
Can you try replacing the iconUrl with `iconUrl: 'public/assets/icon/icon-marker2.svg'` ?

**Comment by rs2-davebanguilan** [comment]
Hi @DwieDima thanks for the reply, Unfortunately it doesn't work on my app. I also tried placing an invalid URL hoping that it will throw an error so I could track it but still, nothing. I also tried targeting it from the `src` folder like: `src/main/assets/public/assets/icon/icon-marker2.svg` but this also doesn't work.

**Comment by jcesarmobile** [comment]
I confirm that it's not implemented for iOS nor Android, but not documented that it's not supported, so I'm marking this as a bug, either to document it or to implement it.

But note that the native Google Maps SDKs don't support using svgs on the markers on iOS nor Android, so if it's implemented it would need to use another format.

**Comment by blinks32** [comment]
I was able to do a fix for android, so the icon can display with a url.

After opening the app in android studio, add this in your _MainActivity.java_  file


```
package io.ionic.starter;
import android.os.Bundle;
import android.os.StrictMode;
import com.getcapacitor.BridgeActivity;
public class MainActivity extends BridgeActivity {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
    StrictMode.setThreadPolicy(policy);
  }
}
```


Then in your list of capacitor plugins select **capacitor-google-maps** Then in the **java** folder select **CapacitorGoogleMapMarker**

Then replace this

```
fun getMarkerOptions(): MarkerOptions {
      val markerOptions = MarkerOptions()
      markerOptions.position(coordinate)
      markerOptions.title(title)
      markerOptions.snippet(snippet)
      markerOptions.alpha(opacity)
      markerOptions.flat(isFlat)
      markerOptions.draggable(draggable)
 return markerOptions
    }
```


With this


```
private fun getBitmapFromURL(src: String?): Bitmap? {
    return try {
      val url = URL(src)
      return BitmapFactory.decodeStream(url.openConnection().getInputStream())
    } catch (e: IOException) {
      e.printStackTrace()
      null
    }
  }

  fun getMarkerOptions(): MarkerOptions {
      val markerOptions = MarkerOptions()
      markerOptions.position(coordinate)
      markerOptions.title(title)
      markerOptions.snippet(snippet)
      markerOptions.alpha(opacity)
      markerOptions.flat(isFlat)
      markerOptions.draggable(draggable)
      markerOptions.icon(getBitmapFromURL(iconUrl)?.let { BitmapDescriptorFactory.fromBitmap(it) })
      return markerOptions
    }

```
As seen below

![Screen Shot 2022-08-24 at 8 02 51 AM](https://user-images.githubusercontent.com/7928001/186353903-ed958275-376b-4080-8a2f-d4d821af25c8.png)


Here is the result

![20220824_075734](https://user-images.githubusercontent.com/7928001/186355074-e07898ac-ef11-4afc-9689-560d828be10d.png)


### I'm open for any jobs, see my profile for more.

**Comment by dev-MMV** [comment]
> I'm open for any jobs, see my profile for more.


I confirm that it works. But it doesn't work with local url.

**Comment by dev-MMV** [comment]
for fix in ios https://github.com/ionic-team/capacitor-plugins/issues/1025

**Comment by jcesarmobile** [comment]
Support for iconUrl was implemented on version 4.2.0 of @capacitor/google-maps.

Only supports local assets, the path needs to be relative to the public folder.
SVG files are not supported (native limitation of the SDK)
